Notifier-service/src/models/conference.ts

```diff
@@ -68,50 +64,56 @@ export class ConferenceModelI extends ConferenceModel{
         return result
     }
 
-    /*
-        TO-DO
-    */
-    async getConferences(offset:number , range:number):Promise<ConferenceDocument[] | null> {
+    async getConferences(offset: number, range: number): Promise<ConferenceDocument[] | null> {
+
         let result = this.makeQuery((model) => {
-            return new Promise<ConferenceDocument[] | null>( (resolve , reject) => {
-                model.find({}).limit(range).exec((err , res) => {
-                    if(!err){
-                        resolve(res);
-                    }
-                    else{
-                        reject(err);
-                    }
-                })
+            return new Promise<ConferenceDocument[] | null>((resolve, reject) => {
+                model.find({ "deadline": { $gte: new Date() } })
+                     .sort({ 'deadline': 1 }).skip(offset).limit(range).exec((err, res) => {
+                            if (!err) {
+                                resolve(res);
+                            }
+                            else {
+                                reject(err);
+                            }
+                     })
             })
         })
-
-        return result;
+        return result
     }
 
-    /*
-        TO-DO
-    */
-    async getConferencesFromCategory(category:string , offset:number , range:number):Promise<ConferenceDocument[] | null> {
-        return Promise.resolve(null);
+    async getConferencesFromCategory(category: string, offset: number, range: number): Promise<ConferenceDocument[] | null> {
+        let result = this.makeQuery((model) => {
+                return new Promise<ConferenceDocument[] | null>((resolve, reject) => {
+                    model.find({ 'categories': { $in: [category] } }, (err, res) => {
+                        if (!err) {
+                            resolve(res);
+                        }
+                        else {
+                            reject(err);
+                        }
+                    }).skip(offset).limit(range);
```

Comment:
* No changes here ? Suggested to change it as above.

Reply:
should i replace this one also with exec()?